pkg/datapath/iptables/iptables.go

```diff
@@ -761,9 +765,60 @@ func ciliumXfrmRules(prog, input string) error {
 	return nil
 }
 
-func addCiliumXfrmRules() error {
+// Exclude crypto traffic from the filter and nat table rules.
+// This avoids encryption bits and keyID, 0x*d00 for decryption
+// and 0x*e00 for encryption, colliding with existing rules. Needed
+// for kube-proxy for example.
+func addCiliumAcceptXfrmRules() error {
+	if option.Config.EnableIPSec == false {
+		return nil
+	}
+	insertAcceptXfrm := func(table, chain string) error {
+		matchFromIPSecEncrypt := fmt.Sprintf("%#08x/%#08x", linux_defaults.RouteMarkDecrypt, linux_defaults.RouteMarkMask)
+		matchFromIPSecDecrypt := fmt.Sprintf("%#08x/%#08x", linux_defaults.RouteMarkEncrypt, linux_defaults.RouteMarkMask)
+
+		comment := "exclude xfrm marks from " + table + " " + chain + " chain"
+
+		if err := runProg("iptables", []string{
+			"-t", table,
+			"-A", chain,
+			"-m", "mark", "--mark", matchFromIPSecEncrypt,
+			"-m", "comment", "--comment", comment,
+			"-j", "ACCEPT"}, false); err != nil {
```

Comment:
Are these rules guaranteed to be hit before the kube-proxy rules? How?

Reply:
They are in the CILIUM_* chains which before kube-proxy rules in the systems I'm running on. If CILIUM_* chains are run after other rules in iptables don't we risk not hitting our rules in other cases as well?

I assume its just order of operations that gets the ordering so that Cilium rules run first.

Reply:
@tgraf Looked into this a bit more as far as I can tell there is no guarantee here other than convention that Cilium rules are installed first. However, on my all my systems (GKE, self-managed, and vagrant) the chains are as follow,
```
Chain OUTPUT (policy ACCEPT)                                                                                                                                              
target     prot opt source               destination                                                                                                                      
CILIUM_OUTPUT  all  --  anywhere           anywhere        
KUBE-SERVICES  all  --  anywhere             anywhere             ctstate NEW
KUBE-FIREWALL  all  --  anywhere             anywhere                                                     
```
I think to go forward its best to apply the fix here, document the possibility of collisions if cilium rules are not on top of list in the troubleshooting part of encryption docs, and then build something to detect this case and repair in another PR.